### PR TITLE
Docs/developing instructions

### LINKED ISSUE #146: Add instructions for developers
@phackstock added the GitHub Actions workflow for publishing a release on pypi in #90, but did not add explicit instructions for (new) developers.

I suggest to add a `DEVELOPING.rst` file using the "Release" section of the developer-notes of the iam-units package as a starting point (see [here](https://github.com/IAMconsortium/units/blob/main/DEVELOPING.rst)). The instructions should be pretty much identical between the two packages (but good to check first while doing an actual release of **nomenclature**).

### PR DESCRIPTION
**Pull request by luciecastella**
Closes #146 

I realized that I cannot create a release myself for authorization reasons so I will not be able to check that these are the right steps. @phackstock when #161 is merged, maybe we can look together into the release and see if anythings is missing from my procedure?

### CODE REVIEW

**Comment by phackstock** [comment]
Ah yes sure, we can go through the steps together.

**Comment by phackstock** [comment]
@danielhuppmann, @luciecastella and I went through the release procedure together and updated the document. We would like to include this PR into the upcoming release so a quick look from your side would be much appreciated.

File: DEVELOPING.rst [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+that the push and scheduled builds are passing. Address any failures before releasing.
+
+1. Test if the build runs locally, by running ``python -m build --sdist --wheel --outdir dist/``.
+   Fix any errors by creating a PR.
+
+2. Tag the release candidate (RC) version on the main branch locally after fixing any 
+   packaging issues, with a ``rc<N>`` suffix, and push to upstream::
+
+    $ git tag v<release version>rc<N>>
+    $ git push upstream v<release version>rc<N>>
```
Comment (danielhuppmann) [inline DEVELOPING.rst:15]:
```suggestion
    $ git push upstream v<release version>rc<N>
```

File: DEVELOPING.rst [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+   and that the release candidate was successfully uploaded to TestPyPI. The address 
+   will be https://test.pypi.org/project/nomenclature-iamc/<release version>rc<N>/. 
+   E.g.: https://test.pypi.org/project/nomenclature-iamc/0.5rc1/ for v0.5rc1.
+
+4. (Optional) Create a fresh virtual environment, download the release from TestPyPi and
+   check that tests are passing.
+   In order to install correctly from TestPypi use the following, (adjust the version pin
+   in the end accordingly)::
+   
+    $ pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple nomenclature-iamc==0.5rc1
```
Comment (danielhuppmann) [inline DEVELOPING.rst:27]: Consistency with the code snippets above...
```suggestion
    $ pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple nomenclature-iamc==v<release version>rc<N>
```

File: DEVELOPING.rst [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+5. Visit https://github.com/IAMconsortium/nomenclature/releases and mark the new release
+   by creating the tag and release simultaneously. The name of the tag is 
+   v<release version> (without the rc<N>).
+
+6. Check at https://github.com/IAMconsortium/nomenclature/actions/workflows/publish.yaml
```
Comment (danielhuppmann) [inline DEVELOPING.rst:33]: Please be consistent about whether to use names or url's for the GitHub Actions

**Review by danielhuppmann — COMMENTED** [review]
A few suggestions inline...

**Comment by phackstock** [comment]
Addressed your comments @danielhuppmann. In the process I also fixed the formatting, you can check only the non-formatting related changes here: https://github.com/IAMconsortium/nomenclature/pull/162/commits/3adc5a57ae65e45064389c0763c65f3b22c27c9f.

**Review by danielhuppmann — APPROVED** [review]
Thank you @luciecastella & @phackstock!

**Pull request merged**